Update Next Steps with more up-to-date information

Menardi opened the pull request:
<!--
Please make sure the checklist boxes are all checked before submitting the PR. The checklist
is intended as a quick reference, for complete details please see our Contributor Guidelines:

http://cordova.apache.org/contribute/contribute_guidelines.html

Thanks!
-->

### Platforms affected
Docs

### What does this PR do?
Recently I was reading the `Next Steps` page on the docs, and noticed it was very outdated. For example, it references things like iOS 6 and Cordova 3 as if they are brand new. I have updated the content to remove outdated information, and bring other stuff up to date.

I have also removed links to jQuery Mobile, as it is no longer really being maintained. I personally don't think we should be encouraging people in 2018 to make jQuery Mobile apps! I have also added links to Vue, and Onsen UI (full disclosure, I am a maintainer of Onsen UI).

### What testing has been done on this change?
Previewed locally in browser.

### Checklist
- [ ] Commit message follows the format: "GH-3232: (android) Fix bug with resolving file paths", where GH-xxxx is the GitHub issue ID & "android" is the platform affected.
- [ ] Added automated test coverage as appropriate for this change.

dpogue commented on www/docs/en/dev/guide/next/index.md:
```diff
@@ -23,85 +23,62 @@ description: A look at topics that new Cordova developers will encounter.
 Consider the following issues to improve the performance in your mobile applications:
 
-**Click versus Touch** - The biggest and simplest mistake you can make is to use click events. While these "work" just fine on mobile, most devices impose a 300ms delay on them in order to distinguish between a touch and a touch "hold" event. Using `touchstart`, or `touchend`, will result in a dramatic improvement - 300ms doesn't sound like much, but it can result in jerky UI updates and behavior ...
+**Click versus Touch** - Many devices impose a 300ms delay on click events in order to distinguish between a click and double click event. This can have the effect of making your app feel slow and unresponsive. Avoiding this delay is one of the most important ways of improving your app's perceived performance. Most Android versions no longer impose this delay, but iOS by default still does. For i ...
```
Maybe worth mentioning that this depends on the value in a viewport tag (in fact, maybe the viewport meta tag should be its own heading in here)

Menardi replied:
I forgot about the need for the `meta` tag, I'll add a bit about that.

dpogue commented on www/docs/en/dev/guide/next/index.md:
```diff
@@ -23,85 +23,62 @@ description: A look at topics that new Cordova developers will encounter.
 ## Upgrading Cordova Projects
 
-If your existing project was created using Cordova 3.x, you can upgrade the project by issuing the following:
+You can upgrade the project by running the following:
```
I don't know that this command actually works... I seem to recall that it either renames the platform directory and installs a newer version beside it, or that it tries to actually upgrade the platforms files and fails horribly.

dpogue commented on www/docs/en/dev/guide/next/index.md:
```diff
@@ -114,80 +91,77 @@ cordova plugin add "some-plugin"
+It is, of course, impossible to test on every possible device on the market. For this reason, it's wise to recruit many testers who have different devices. Although they won't catch every problem, chances are good that they will discover quirks and issues that you would never find alone.
 
-Tip: It is possible on Android Nexus devices to easily flash different versions of Android onto the device. This simple process will allow you to easily test your application on different levels of Android with a single device, without voiding your warranty or requiring you to “jailbreak” or “root” your device. Refer to the [Google Android factory images and instructions](https://developers.googl ...
+Tip: It is possible on Google Pixel devices to easily flash different versions of Android onto the device. This simple process will allow you to easily test your application on different levels of Android with a single device, without voiding your warranty or requiring you to “jailbreak” or “root” your device. Refer to the [Google Android factory images and instructions](https://developers.google ...
```
I would remove this bit entirely.

The Android emulator has improved ***significantly*** in the past few years, to the point I suspect people are better off using that for basic testing as opposed to flashing different versions onto an expensive phone.

Menardi commented:
Thanks for the review, @dpogue. I've updated the PR to remove the things you mentioned, and also added info about needing the meta tag to prevent the 300ms tap delay.
